ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/LedgerApiServer.scala

```diff
@@ -52,136 +65,99 @@ object LedgerApiServer {
-          desiredPort,
-          maxInboundMessageSize,
-          address,
-          loggerFactory,
-          sslContext,
-          interceptors,
-          metrics
-        )
-
-        /** returns the api port the server is listening on */
-        override def port: Int = impl.port
-
-        /** returns when all services have been closed during the shutdown */
-        override def servicesClosed(): Future[Unit] = impl.servicesClosed()
-
-        override def close(): Unit = {
-          impl.close()
-          serverEsf.close()
+      val builder = address.fold(NettyServerBuilder.forPort(desiredPort))(address =>
+        NettyServerBuilder.forAddress(new InetSocketAddress(address, desiredPort)))
+      sslContext
+        .fold {
+          logger.info("Starting plainText server")
+        } { sslContext =>
+          logger.info("Starting TLS server")
+          val _ = builder.sslContext(sslContext)
         }
+      builder.directExecutor()
+      builder.channelType(classOf[NioServerSocketChannel])
+      builder.bossEventLoopGroup(bossEventLoopGroup)
+      builder.workerEventLoopGroup(workerEventLoopGroup)
+      builder.permitKeepAliveTime(10, SECONDS)
+      builder.permitKeepAliveWithoutCalls(true)
+      builder.maxInboundMessageSize(maxInboundMessageSize)
+      interceptors.foreach(builder.intercept)
+      builder.intercept(new MetricsInterceptor(metrics))
+      apiServices.services.foreach(builder.addService)
+      val grpcServer = builder.build()
+
+      try {
+        grpcServer.start()
+      } catch {
+        case io: IOException if io.getCause != null && io.getCause.isInstanceOf[BindException] =>
+          stop()
+          throw new UnableToBind(desiredPort, io.getCause)
+        case NonFatal(e) =>
+          stop()
+          throw e
       }
-    }(mat.executionContext)
-  }
+      closeables.push(() => {
+        grpcServer.shutdown()
+        if (!grpcServer.awaitTermination(10, SECONDS)) {
+          logger.warn(
+            "Server did not terminate gracefully in one second. " +
+              "Clients probably did not disconnect. " +
+              "Proceeding with forced termination.")
+          val _ = grpcServer.shutdownNow()
+        }
+      })
 
-}
+      val host = address.getOrElse("localhost")
+      val actualPort = grpcServer.getPort
+      logger.info(s"listening on $host:$actualPort")
+
+      val servicesClosedP = Promise[Unit]()
+      closeables.push(() => {
+        apiServices.close()
+        servicesClosedP.success(())
+      })
 
-private class LedgerApiServer(
-    apiServices: ApiServices,
-    desiredPort: Int,
-    maxInboundMessageSize: Int,
-    address: Option[String],
-    loggerFactory: NamedLoggerFactory,
-    sslContext: Option[SslContext] = None,
-    interceptors: List[ServerInterceptor] = List.empty,
-    metrics: MetricRegistry)(implicit mat: ActorMaterializer)
-    extends ApiServer {
+      new ApiServer {
+        override def port: Int = actualPort
```

Comment:
```suggestion
        override val port: Int = actualPort
```
Looks like a `val` is enough here.

Reply:
Done.